Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModuleEditor.cs

```diff
@@ -127,7 +127,22 @@ public override void OnInspectorGUI()
                 if (m_AvailableActionsInAsset == null)
                     continue;
 
-                var index = Array.IndexOf(m_AvailableActionsInAsset, m_ReferenceProperties[i].objectReferenceValue) + 1;
+                // find the input action reference from the asset that matches the input action reference from the
+                // InputSystemUIInputModule that is currently selected. Note we can't use reference equality of the
+                // two InputActionReference objects here because in ReassignActions above, we create new instances
+                // every time it runs.
+                var index = 0;
+                var inputAction = ((InputActionReference)m_ReferenceProperties[i]?.objectReferenceValue)?.action;
```

Comment:
Just a thought, and minor, but if objectReferenceValue and action here are both possible to be null, maybe skip the whole for-loop thing if inputAction is null?

Reply:
Done